File: frontend/packages/shared/src/types/army.ts
export type TroopType = 'hc' | 'lc' | 'hi' | 'li' | 'ar' | 'ma';

export type MaterialRank = 'none' | 'wood' | 'leather' | 'bronze' | 'steel' | 'mithril';

export type ArmyDirection = 'h' | 'ne' | 'e' | 'se' | 'sw' | 'w' | 'nw';

export type Tactics = 'ch' | 'fl' | 'st' | 'su' | 'hr' | 'am';

export interface Army {
  id: string;
  nationId: string;
  name: string;
  locationHex: string;
  commanderId?: string;
  heavyCavalry: number;
  lightCavalry: number;
  heavyInfantry: number;
  lightInfantry: number;
  archers: number;
  menAtArms: number;
  weaponRank: number;
  armourRank: number;
  morale: number;
  training: number;
  food: number;
  warMachines: number;
  isOnManoeuvres: boolean;
  size: number;
}

export interface Navy {
  id: string;
  nationId: string;
  warships: number;
  transports: number;
  locationHex: string;
  commanderId?: string;
  strength: number;
}

export const TROOP_COSTS: Record<TroopType, number> = {
  hc: 3000,
  lc: 1500,
  hi: 2000,
  li: 1000,
  ar: 1000,
  ma: 500,
};

export const MATERIAL_RANKS: Record<MaterialRank, number> = {
  none: 0,
  wood: 10,
  leather: 20,
  bronze: 40,
  steel: 60,
  mithril: 100,
};

export const FOOD_CONSUMPTION: Record<TroopType, number> = {
  hc: 2,
  lc: 3,
  hi: 1,
  li: 1,
  ar: 1,
  ma: 1,
};

File: backend/MEPBMmanager.Domain/Entities/Army.cs
namespace MEPBMmanager.Domain.Entities;

public class Army
{
    public string Id { get; set; } = string.Empty;
    public string NationId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string LocationHex { get; set; } = string.Empty;
    public string? CommanderId { get; set; }
    
    // Troops
    public int HeavyCavalry { get; set; } = 0;
    public int LightCavalry { get; set; } = 0;
    public int HeavyInfantry { get; set; } = 0;
    public int LightInfantry { get; set; } = 0;
    public int Archers { get; set; } = 0;
    public int MenAtArms { get; set; } = 0;
    
    // Equipment per troop type
    public int HCWeaponRank { get; set; } = 10;
    public int HCArmourRank { get; set; } = 0;
    public int LCWeaponRank { get; set; } = 10;
    public int LCArmourRank { get; set; } = 0;
    public int HIWeaponRank { get; set; } = 10;
    public int HIArmourRank { get; set; } = 0;
    public int LIWeaponRank { get; set; } = 10;
    public int LIArmourRank { get; set; } = 0;
    public int ArcherWeaponRank { get; set; } = 10;
    public int ArcherArmourRank { get; set; } = 0;
    public int MAAWeaponRank { get; set; } = 10;
    public int MAAArmourRank { get; set; } = 0;
    
    // Stats
    public int Morale { get; set; } = 30;
    public int Training { get; set; } = 10;
    public int Food { get; set; } = 0;
    public int WarMachines { get; set; } = 0;
    public bool IsOnManoeuvres { get; set; } = false;
    
    public Nation Nation { get; set; } = null!;
    public ICollection<Character> Characters { get; set; } = new List<Character>();
    public ICollection<Order> Orders { get; set; } = new List<Order>();
    public ICollection<Encounter> Encounters { get; set; } = new List<Encounter>();
}
